### ISSUE TITLE
Bug with grounding GlobNodes

### ISSUE THREAD

**Issue by AmeBel** [opencog/atomspace#968]
### Content of test.scm
```scheme
(use-modules (opencog))
(use-modules (opencog atom-types))
(use-modules (opencog exec))

(define input-utterance-words (AnchorNode "Chatbot: InputUtteranceWords"))

(StateLink
    input-utterance-words
    (ListLink
        (Word "Han")
        (Word "now")
        (Word "let")
        (Word "us")
        (Word "show")
        (Word "off")
        (Word "reasoning")
    )
)


(define (test)
    (cog-execute! (Get
        (State
            input-utterance-words
            (List
                (Glob "Robot")
                (Word "let")
                (Word "us")
                (Word "show")
                (Glob "Mode")
            )
        )
    ))
)
```

### Test run 
```scheme
scheme@(guile-user)> (load "test.scm")
scheme@(guile-user)> (test)
$1 = (SetLink
)
```

### Expected output
```scheme
(SetLink 
   (ListLink 
      (ListLink
         (WordNode "Han")     
         (WordNode "now")       
      )  
      (ListLink
         (WordNode "off")    
         (WordNode "reasoning")   
      )  
   )
)  
```

To get the expected output run https://github.com/hansonrobotics/HEAD by executing `./dev.sh --oc`
and telnet to the cogserver scheme shell.
```scheme
; From the scheme shell on cogserver port 17020
guile> (load "/path/to/test.scm")
guile> (test)
(SetLink
   (ListLink
      (ListLink
         (WordNode "Han")
         (WordNode "now")
      )
      (ListLink
         (WordNode "off")
         (WordNode "reasoning")
      )
   )
)

guile> (cog-incoming-set input-utterance-words)
((StateLink
   (AnchorNode "Chatbot: InputUtteranceWords")
   (ListLink
      (WordNode "Han")
      (WordNode "now")
      (WordNode "let")
      (WordNode "us")
      (WordNode "show")
      (WordNode "off")
      (WordNode "reasoning")
   )
)
 (StateLink
   (AnchorNode "Chatbot: InputUtteranceWords")
   (ListLink
      (GlobNode "Robot")
      (WordNode "let")
      (WordNode "us")
      (WordNode "show")
      (GlobNode "Mode")
   )
)
)
```

**Comment by linas** [comment]
yep its a bug. I'll look at it after I am done with #940

**Comment by linas** [comment]
.. unless someone else wants to look at it. The glob matching code is mostly all just in one place in one file, instead of being a complicated interaction of parts scattered across many files. So it should be relatively "easy" to debug. The code is here:   https://github.com/opencog/atomspace/blob/master/opencog/query/PatternMatchEngine.cc#L215-L278 

Unless... Hmm. Did this used to work, say, a week ago? Because my recent commits might have wreked this, not sure.

**Comment by AmeBel** [comment]
Resolved by #971 .
Thanks @linas
